apps/drec-api/src/pods/issuer/issuer.service.ts

```diff
@@ -0,0 +1,172 @@
+  async handleCron(): Promise<void> {
+    const deviceGroupRule1 =
+      this.configService.get<string>('DEVICE_GROUP_RULE_1') ||
+      'registrant_organisation_code';
+    const deviceGroupRule2 =
+      this.configService.get<string>('DEVICE_GROUP_RULE_2') || 'country_code';
+    this.logger.debug('Called every day at 23:30 Server time');
+
+    const startDate = DateTime.now().minus({ days: 1 }).toUTC();
+    const endDate = DateTime.now().minus({ minute: 1 }).toUTC();
+
+    this.logger.debug(`Start date ${startDate} - End date ${endDate}`);
+
+    const groups = await this.groupService.getAll();
+    await Promise.all(
+      groups.map(async (group: DeviceGroup) => {
+        group.devices = await this.deviceService.findForGroup(group.id);
+        return await this.issueCertificateForGroup(group, startDate, endDate);
+      }),
+    );
+    const devicesWithoutGroups = await this.deviceService.findMultiple({
+      where: {
+        groupId: null,
+      },
+    });
+
+    if (devicesWithoutGroups.length) {
+      const ownerGroupedDevices = this.groupBy(
+        devicesWithoutGroups,
+        deviceGroupRule1,
+      );
+
+      await Promise.all(
+        ownerGroupedDevices?.map(async (ownerBasedGroup: Device[], i) => {
+          this.groupBy(ownerBasedGroup, deviceGroupRule2)?.map(
+            async (countryBasedGroup: Device[], j) => {
+              const categorizedGroup: IDeviceGroup = {
+                id: 0,
+                name: `${ownerBasedGroup[i][deviceGroupRule1]}_${countryBasedGroup[j][deviceGroupRule2]}`,
+                organizationId: ownerBasedGroup[i].registrant_organisation_code,
+                devices: countryBasedGroup,
+              };
+              return await this.issueCertificateForGroup(
+                categorizedGroup,
+                startDate,
+                endDate,
+              );
+            },
+          );
+        }),
+      );
+    }
+  }
+
+  private async issueCertificateForGroup(
+    group: IDeviceGroup,
+    startDate: DateTime,
+    endDate: DateTime,
+  ): Promise<void> {
+    const readsFilter: FilterDTO = {
+      offset: 0,
+      limit: 1000,
+      start: startDate.toString(),
+      end: endDate.toString(),
+    };
+
+    const org = await this.organizationService.findOne(group.organizationId);
+    if (!org) {
+      throw new NotFoundException(
+        `No organization found with code ${group.organizationId}`,
+      );
+    }
+    const groupReads: number[] = [];
+    await Promise.all(
+      group.devices.map(async (device: Device) =>
+        groupReads.push(await this.getDeviceFullReads(device.id, readsFilter)),
+      ),
+    );
+    const totalReadValue = groupReads.reduce(
+      (accumulator, currentValue) => accumulator + currentValue,
+      0,
+    );
+    // Convert from W to kW
+    const totalReadValueKw = Math.round(totalReadValue * 10 ** -3);
+    console.log('totalReadValue: ', totalReadValue);
```

Comment:
hmm

Reply:
Yeah, will remove all console.log before merging

Reply:
You should remove them before reviews. No code changes should happen between the last review approval and merging.

Reply:
Yep..those I forgot. I meant to say that I will check for all console.logs left out.